hazelcast/src/main/java/com/hazelcast/nio/Packet.java

```diff
@@ -38,26 +38,56 @@
 
     public static final byte VERSION = 4;
 
+
+    //             PACKET HEADER FLAGS
+    //
+    // Flags are dispatched against in a cascade:
+    // 1. URGENT (bit 4)
+    // 2. Packet type (bits 0, 2, 5, 7)
+    // 3. Flags specific to a given packet type (bits 1, 6)
+
+
+    // 1. URGENT flag
+
+    /** Marks the packet as Urgent  */
+    public static final int FLAG_URGENT = 1 << 4;
+
+
+    // 2. Packet type flags, mutually exclusive
+
+    /** Marks the packet type as Operation */
     public static final int FLAG_OP = 1 << 0;
-    public static final int FLAG_RESPONSE = 1 << 1;
+    /** Marks the packet type as Event */
     public static final int FLAG_EVENT = 1 << 2;
-    public static final int FLAG_URGENT = 1 << 4;
+    /** Marks the packet type as Bind message */
     public static final int FLAG_BIND = 1 << 5;
+    /** Marks the packet as Jet */
+    public static final int FLAG_JET = 1 << 7;
 
-    /**
-     * A flag to indicate this is a special control packet for the operation system like invocation-heartbeats.
-     */
+
+    // 3. Type-specific flags. Same bits can be reused within each type
+
+    // 3.a Operation packet flags
+
+    /** Marks an Operation packet as Response */
+    public static final int FLAG_OP_RESPONSE = 1 << 1;
```

Comment:
does it have to be an operation? right now operation is the only type which can generate a response, but there is nothing preventing you to introduce a new packet type yielding a response.